packages/qiskit/quri_parts/qiskit/backend/connectivity.py

```diff
@@ -0,0 +1,22 @@
+import networkx as nx
+from qiskit.providers.ibmq import IBMQBackend
+
+
+def device_connectivity_graph(device: IBMQBackend) -> nx.Graph:
```

Comment:
I wonder if we can use this function with a backend more general than `IBMQBackend`.
In `QiskitSamplingBackend`, `BackendV1` is used when accessing `backend.configuration()`:
https://github.com/QunaSys/quri-parts/blob/4e739e0fd7e21f4734e4cc742dec97ce194774b8/packages/qiskit/quri_parts/qiskit/backend/sampling.py#L122
For `BackendV2`, it seems that `num_qubits` and `coupling_map` can be retrieved as direct attributes:
https://qiskit.org/documentation/stubs/qiskit.providers.BackendV2.html#backendv2

Reply:
Would you like me to add support for both `BackendV1` and `BackendV2`? 

### `BackendV1` Fetching

The `BackendV1` class has a `configuration` method that returns a `BackendConfiguration` ([link](https://qiskit.org/documentation/_modules/qiskit/providers/models/backendconfiguration.html#BackendConfiguration)). This object contains `num_qubits` and `coupling_map` attributes (refer to source [code](https://qiskit.org/documentation/_modules/qiskit/providers/models/backendconfiguration.html#BackendConfiguration)). `coupling_map` is of the type `CouplingMap` ([documentation](https://qiskit.org/documentation/stubs/qiskit.transpiler.CouplingMap.html)) with a `get_edges` method that returns the list of edges in the map. The required connectivity graph that we use can be constructed from this.


### `BackendV2` Fetching

The `BackendV2` class has a much simpler interface. I can use the `coupling_map` and `num_qubits` attributes directly and construct the required graph.

By doing instance checking, I could handle both of these cases. For testing purposes, I will modify the classes I have constructed to match the backend functionality in qiskit.